### DISCUSSION TITLE
Primary Key Type

### DISCUSSION THREAD

**Discussion by mickeyl** [groue/GRDB.swift#1189] [General]
All examples I have seen seem to use an `Int64` as `autoIncrementedPrimaryKey`. Is that a necessity induced by SQLite or just the way GRDB usually works? Could we just use a column with a unique, say, `String`?

**Comment by groue** [comment]
Hello @mickeyl,

Both SQLite and GRDB supports all primary keys, including composite primary keys that span multiple columns. It just happens that auto-incremented integer primary keys are a common use case, and this is why those are frequently met in sample codes. You'll meet other kinds of primary keys as well.

To learn more about SQLite primary keys: check out https://www.sqlite.org/lang_createtable.html#the_primary_key

Design your SQLite primary keys freely: GRBD will just make it work.
